ksql-engine/src/test/java/io/confluent/ksql/function/UdfCompilerTest.java

```diff
@@ -130,6 +200,91 @@ public void shouldThrowKsqlFunctionExceptionIfNullPassedWhenExpectingPrimitiveTy
     udf.eval(this, new Object[]{null});
   }
 
+
+  @Test
+  public void shouldThrowWhenUdafReturnTypeIsntAUdaf() throws NoSuchMethodException {
+    expectedException.expect(KsqlException.class);
+    expectedException.expectMessage("UDAFs must implement io.confluent.ksql.function.udaf.Udaf "
+        + "or io.confluent.ksql.function.udaf.TableUdaf .method='createBlah', functionName='test'"
+        + " UDFClass='class io.confluent.ksql.function.UdfCompilerTest");
+    udfCompiler.compileAggregate(UdfCompilerTest.class.getMethod("createBlah"),
+        classLoader,
+        "test"
+    );
+  }
+
+  @Test
+  public void shouldHandleUdafsWithLongValTypeDoubleAggType() throws NoSuchMethodException {
+    udfCompiler.compileAggregate(UdfCompilerTest.class.getMethod("createLongDouble"),
+        classLoader,
+        "test"
+    );
+  }
+
+  @Test
+  public void shouldHandleUdafsWithDoubleValTypeLongAggType() throws NoSuchMethodException {
+    udfCompiler.compileAggregate(UdfCompilerTest.class.getMethod("createDoubleLong"),
+        classLoader,
+        "test"
+    );
+  }
+
+  @Test
+  public void shouldHandleUdafsWithIntegerValTypeStringAggType() throws NoSuchMethodException {
+    udfCompiler.compileAggregate(UdfCompilerTest.class.getMethod("createIntegerString"),
+        classLoader,
+        "test"
+    );
+  }
+
+  @Test
+  public void shouldHandleUdafsWithStringValTypeIntegerAggType() throws NoSuchMethodException {
+    udfCompiler.compileAggregate(UdfCompilerTest.class.getMethod("createStringInteger"),
+        classLoader,
+        "test"
+    );
+  }
+
+  @Test
+  public void shouldHandleUdafsWithBooleanValTypeListAggType() throws NoSuchMethodException {
+    udfCompiler.compileAggregate(UdfCompilerTest.class.getMethod("createBooleanList"),
+        classLoader,
+        "test"
+    );
+  }
+
+  @Test
+  public void shouldHandleUdafsWithListValTypeBooleamAggType() throws NoSuchMethodException {
+    udfCompiler.compileAggregate(UdfCompilerTest.class.getMethod("createListBoolean"),
+        classLoader,
+        "test"
+    );
+  }
+
+  @Test
+  public void shouldHandleUdafsWithMapValMapAggTypes() throws NoSuchMethodException {
+    udfCompiler.compileAggregate(UdfCompilerTest.class.getMethod("createMapMap"),
+        classLoader,
+        "test"
+    );
+  }
+
+  @Test(expected = KsqlException.class)
```

Comment:
Similar to the comment above, it would be good to get a sense of the exception message returned to the user here. It needs to be actionable.